apps/microtvm/zephyr/template_project/microtvm_api_server.py

```diff
@@ -59,44 +60,13 @@
 
 # Data structure to hold the information microtvm_api_server.py needs
 # to communicate with each of these boards.
-BOARD_PROPERTIES = {
-    "qemu_x86": {
-        "board": "qemu_x86",
-        "model": "host",
-    },
-    "qemu_riscv32": {
-        "board": "qemu_riscv32",
-        "model": "host",
-    },
-    "qemu_riscv64": {
-        "board": "qemu_riscv64",
-        "model": "host",
-    },
-    "mps2_an521": {
-        "board": "mps2_an521",
-        "model": "mps2_an521",
-    },
-    "nrf5340dk_nrf5340_cpuapp": {
-        "board": "nrf5340dk_nrf5340_cpuapp",
-        "model": "nrf5340dk",
-    },
-    "stm32f746g_disco": {
-        "board": "stm32f746g_disco",
-        "model": "stm32f746xx",
-    },
-    "nucleo_f746zg": {
-        "board": "nucleo_f746zg",
-        "model": "stm32f746xx",
-    },
-    "nucleo_l4r5zi": {
-        "board": "nucleo_l4r5zi",
-        "model": "stm32l4r5zi",
-    },
-    "qemu_cortex_r5": {
-        "board": "qemu_cortex_r5",
-        "model": "zynq_mp_r5",
-    },
-}
+BOARD_PROPERTIES = None
+BOARDS_FILE_NAME = "boards.json"
+
+with open(
+    BOARDS_FILE_NAME,
+) as board_f:
+    BOARD_PROPERTIES = json.load(board_f)
```

Comment:
A couple of suggestions here.

I think it's not necessary to enforce `BOARD_PROPERTIES = None` before `with` since it won't create a new scope -- please experiment with that possibility.

I prefer the long form, e.g. 
```
with open(BOARDS) as boards:
    BOARD_PROPERTIES = json.load(boards)
```
I do expect the linter won't complain about it.

I think `with` should be inside a `try` in case the file is missing, corrupt etc for whatever reason, and `except` so could raise a BoardProperties(exception) class like there is already one for `BoardError` and `BoardAutodetectFailed`. I recall @areusch concerned once about overloading too much exceptions in the server side, so I'm wondering what he thinks about that suggestion actually :)

Reply:
since this is used once I think we could just use FileNotFoundError exception? something like this:

```
try:
    with open(BOARDS) as boards:
        BOARD_PROPERTIES = json.load(boards)
except FileNotFoundError:
    raise FileNotFoundError(f"Board file {{{BOARDS}}} does not exist.")
```

Reply:
@mehrdadh yeah I think that's fine too. Btw, do we need triple `{` `}` here?

Reply:
yeah because of f"" statement

Reply:
@mehrdadh ah, I see, you want to have the file name between `{` and `}` too. OK.